node/src/components/network/message.rs

```diff
@@ -418,6 +428,157 @@ pub struct EstimatorWeights {
+            "network_name_limit" => self.chainspec.network_config.name.len() as i64,
+            // These limits are making deploys bigger then they actually are, since many items
+            // have both a `contract_name` and an `entry_point`. We accept 2X as an upper bound.
+            "contract_name_limit" => self.chainspec.deploy_config.max_deploy_size as i64,
+            "entry_point_limit" => self.chainspec.deploy_config.max_deploy_size as i64,
+            "recent_era_count" => {
+                (self.chainspec.core_config.unbonding_delay
+                    - self.chainspec.core_config.auction_delay) as i64
+            }
+            "validator_count" => self.chainspec.core_config.validator_slots as i64,
+            "minimum_era_height" => self.chainspec.core_config.minimum_era_height as i64,
+            "era_duration_ms" => self.chainspec.core_config.era_duration.millis() as i64,
+            "minimum_round_length_ms" => self
+                .chainspec
+                .core_config
+                .minimum_block_time
+                .millis()
+                .max(1) as i64,
+            "module_bytes" => self.chainspec.deploy_config.max_deploy_size as i64,
+            "approvals_hashes" => {
+                (self.chainspec.deploy_config.block_max_deploy_count
+                    + self.chainspec.deploy_config.block_max_transfer_count) as i64
+            }
+            "max_deploys_per_block" => self.chainspec.deploy_config.block_max_deploy_count as i64,
+            "max_transfers_per_block" => {
+                self.chainspec.deploy_config.block_max_transfer_count as i64
+            }
+            "average_approvals_per_deploy_in_block" => {
+                let max_total_deploys = (self.chainspec.deploy_config.block_max_deploy_count
+                    + self.chainspec.deploy_config.block_max_transfer_count)
+                    as i64;
+
+                // Note: The +1 is to overestimate, as depending on the serialization format chosen,
+                //       spreading out the approvals can be increase or decrease the size. For
+                //       example, in a length-prefixed encoding, putting them all in one may result
+                //       in a smaller size if variable size integer encoding it used. In a format
+                //       using separators without trailing separators (e.g. commas in JSON),
+                //       spreading out will reduce the total number of bytes.
+                ((self.chainspec.deploy_config.block_max_approval_count as i64 + max_total_deploys
+                    - 1)
+                    / max_total_deploys)
+                    .max(0)
+                    + 1
+            }
+            "max_accusations_per_block" => self.chainspec.core_config.validator_slots as i64,
+            // `RADIX` from EE.
+            "max_pointer_per_node" => 255,
+            "endorsements_disabled" => 1,
```

Comment:
We're negating this at all callsites, so I'd suggest renaming it to `endorsements_enabled` and setting it to `0`.

Also, it might be worth a code comment explaining why this isn't being derived from a chainspec value?

Reply:
AFAIK it is hard-disabled in the code as a workaround, thus also the inversion. I am assuming either the code will be removed entirely, or the feature re-enabled, both cases obviating the need for the setting.